src/emmodel/data.py

```diff
@@ -0,0 +1,82 @@
+"""
+Data module
+"""
+from dataclasses import dataclass, field
+from typing import List, Tuple, Dict
+
+import numpy as np
+import pandas as pd
+
+
+@dataclass
+class DataProcessor:
+    col_deaths: str
+    col_year: str
+    col_tunit: str
+    col_population: str
+    col_covs: List[str] = field(default_factory=list)
+
+    def __post_init__(self):
+        if "week" in self.col_tunit:
+            self.tunit = "week"
+            self.tunits_per_year = 52
+        elif "month" in self.col_tunit:
+            self.tunit = "month"
+            self.tunits_per_year = 12
+        else:
+            raise ValueError("Unrecognized time unit, must be 'week' or 'month'.")
+
+        self.cols = [self.col_deaths,
+                     self.col_year,
+                     self.col_tunit,
+                     self.col_population] + self.col_covs
```

Comment:
Could col_covs include col_tunit, or col_year, col_population as covariates; if so, then maybe we need to get the unique cols